vivisect/qt/memory.py

```diff
@@ -257,29 +259,41 @@ def makeStructAgainMulti(self, va, parent=None):
 
     @vq_hotkey.hotkey('viv:make:number:one')
     def _hotkey_make_number_one(self):
-        if self._canv_curva != None:
+        if self._canv_curva is not None:
             self.vw.makeNumber(self._canv_curva, 1)
 
     @vq_hotkey.hotkey('viv:make:number:two')
     def _hotkey_make_number_two(self):
-        if self._canv_curva != None:
+        if self._canv_curva is not None:
             self.vw.makeNumber(self._canv_curva, 2)
 
     @vq_hotkey.hotkey('viv:make:number:four')
     def _hotkey_make_number_four(self):
-        if self._canv_curva != None:
+        if self._canv_curva is not None:
             self.vw.makeNumber(self._canv_curva, 4)
 
     @vq_hotkey.hotkey('viv:make:number:eight')
     def _hotkey_make_number_eight(self):
-        if self._canv_curva != None:
+        if self._canv_curva is not None:
             self.vw.makeNumber(self._canv_curva, 8)
 
     @vq_hotkey.hotkey('viv:make:number:sixteen')
     def _hotkey_make_number_sixteen(self):
-        if self._canv_curva != None:
+        if self._canv_curva is not None:
             self.vw.makeNumber(self._canv_curva, 16)
 
+    @vq_hotkey.hotkey('viv:preview:instr')
+    def _hotkey_preview_instr(self):
+        if self._canv_curva is not None:
+            self.vw.previewCode(self._canv_curva)
+
+    @firethread
+    @vq_hotkey.hotkey('viv:save')
+    def _hotkey_save(self, fullsave=False):
```

Comment:
So there's also `_menuFileSave` in the main qt window that does exactly this. This smells like something that should be carved out to a common base class (like the HotKeyMixin class).

Reply:
good catch!  removed.  this code apparently predates when we wrapped ctrl-s saving into mainline :)